core/app/c/[communitySlug]/pubs/[pubId]/components/RelatedPubsTable.tsx

```diff
@@ -0,0 +1,154 @@
+		{
+			header: ({ column }) => <DataTableColumnHeader column={column} title="Title" />,
+			accessorKey: "title",
+			cell: ({ row }) => {
+				const pathname = usePathname();
+				const path = pathname.split("/").slice(0, 4).join("/");
+				const title = getPubTitle(row.original);
+				return (
+					<Link className="block truncate underline" href={`${path}/${row.original.id}`}>
+						{title}
+					</Link>
+				);
+			},
+		},
+		{
+			header: ({ column }) => <DataTableColumnHeader column={column} title="Pub Type" />,
+			accessorKey: "pubType",
+			cell: ({ getValue }) => {
+				const pubTypeName = getValue<PubTypes>().name;
+				return <Badge variant="outline">{pubTypeName}</Badge>;
+			},
+		},
+		{
+			header: ({ column }) => <DataTableColumnHeader column={column} title="Stage" />,
+			accessorKey: "stages",
+			cell: ({ getValue }) => {
+				const stageName = getValue<Stages>()?.name;
+				return stageName ? (
+					<Badge variant="outline">{stageName}</Badge>
+				) : (
+					<span className="text-muted-foreground">None</span>
+				);
+			},
+		},
+		{
+			header: ({ column }) => <DataTableColumnHeader column={column} title="Created At" />,
+			accessorKey: "createdAt",
+			cell: ({ getValue }) => (
+				<time dateTime={new Date().toString()} suppressHydrationWarning>
+					{new Date(getValue<string>()).toLocaleString(undefined, createdAtDateOptions)}
+				</time>
+			),
+		},
+	] as const satisfies ColumnDef<FullProcessedPub, unknown>[];
+};
+
+const Table = ({ pubs }: { pubs: FullProcessedPub[] }) => {
+	const columns = getRelatedPubsColumns();
+
+	return <DataTable columns={columns} data={pubs} hidePaginationWhenSinglePage />;
+};
+
+export const RelatedPubsTable = ({ pub }: { pub: FullProcessedPub }) => {
+	const groupedBySlug = useMemo(() => {
+		const grouped: Record<string, { pub: FullProcessedPub; fieldName: string }[]> = {};
+		for (const value of pub.values) {
+			const { relatedPub, fieldSlug, fieldName } = value;
+			if (relatedPub) {
+				if (!grouped[fieldSlug]) {
+					grouped[fieldSlug] = [];
+				}
+				grouped[fieldSlug].push({ pub: relatedPub, fieldName });
+			}
+		}
+		return grouped;
+	}, [pub]);
+
+	const fields = Object.entries(groupedBySlug).map(([slug, value]) => ({
+		slug,
+		name: value[0].fieldName,
+		count: value.length,
+	}));
+
+	const [selected, setSelected] = useState(fields[0]?.slug);
+
+	if (fields.length === 0) {
+		return <Table pubs={[]} />;
+	}
+
+	return (
+		<div>
+			<div className="flex items-center gap-2">
+				{fields.map((field) => {
+					const isSelected = selected === field.slug;
+					return (
+						<Button
+							key={field.slug}
+							onClick={() => {
+								setSelected(field.slug);
+							}}
+							variant={isSelected ? "default" : "ghost"}
+						>
+							{field.name}
+							<span className="ml-2 font-mono text-xs">{field.count}</span>
+						</Button>
+					);
+				})}
+			</div>
+			<Table pubs={groupedBySlug[selected].map((value) => value.pub)} />
+		</div>
+	);
+};
```

Comment:
even though i theoretically like the `searchParameter` based approach used by the children table more, i have to admit that it's nice how quick navigation is in comparison haha.
I am slightly worried about pages with a ton of related pubs, but we'll just have to solve that with specific pagination for the related pubs i think, so i think this is approach is good!